### PR TITLE
Alternate namespaces #70

### LINKED ISSUE #70: Support deploying into alternate namespaces
**Describe the bug**
Istio doesn't require that you install it to `istio-system`. It can be deployed under another namespace. However the connector/helm is currently hardcoded to `istio-system`.

**Expected behavior**
The helm would expose a template variable allowing the user to override the namespace, and have it apply uniformly to the adapter deployment.

### PR DESCRIPTION
**Pull request by dlinsley**
**Description**
Changes to make namespace configurable for this adapter.

The adapter would not function if the `handler`, `rule` and `adapter` objects were outside of the `istio-system` namespace.  The naming of these objects though is unique enough at the moment to prevent naming conflicts with other Istio adapters.

**Additional context**
This change is functional with `helm template` style of installation.  It will not work with `helm install` since there is more than one namespace involved.

This change works in the event that other Istio adapters are installed on the same Kubernetes cluster and they use the same metric or instance names.  With this change the instance objects can be in a different namespace specific for this adapter.

Feedback welcome!

Fixes #70

### CODE REVIEW

File: install/wavefront/templates/wavefront.yaml [hunk 1]
```diff
@@ -4,11 +4,11 @@ apiVersion: "config.istio.io/v1alpha2"
...
 metadata:
   name: wavefront
-  namespace: istio-system
+  namespace: {{ .Values.namespaces.istio }}
 spec:
   description: 
   session_based: false
   templates:
-  - metric
+  - metric.{{ .Values.namespaces.adapter }}
```
Comment (venilnoronha) [inline install/wavefront/templates/wavefront.yaml]: Could you document the need to update this file in the `BUILD` file? This file is auto-generated by the mixer codegen utility, and we now need to update the namespace sections as a post-generation step.

Ideally, we should revert the change to this file and the `namespaces.istio` value and we should only take the `namespaces.adapter` value as input. But, that's for another day as it requires an additional debugging/update of Istio dependencies.

File: install/wavefront/templates/handler.yaml [hunk 2]
```diff
@@ -24,20 +24,20 @@ spec:
...
-  {{- template "rule.http" }}
+  {{- template "rule.http" . }}
 {{- end }}
 ---
 {{- if eq .Values.metrics.tcp true }}
-  {{- template "instances.tcp" }}
+  {{- template "instances.tcp" . }}
 ---
-  {{- template "rule.tcp" }}
+  {{- template "rule.tcp" . }}
```
Comment (venilnoronha) [inline install/wavefront/templates/handler.yaml]: Unsure about this, do we definitely need the `.` scoping?

File: install/wavefront/README.md [hunk 3]
```diff
@@ -94,7 +94,8 @@ See below for the list of available configuration parameters.
 |             | source        | The source tag for all metrics                     |
 |             | prefix        | The prefix to prepend all metrics with             |
 |             | http          | Specify whether HTTP metrics should be captured    |
-|             | tcp           | Specify whether TCP metrics should be captured     |
```
[the remark this answers is not in the document]
Comment (dlinsley) [reply install/wavefront/README.md]: Yes, will fix

File: install/wavefront/templates/handler.yaml (hunk 2 above)
Comment (dlinsley) [reply install/wavefront/templates/handler.yaml]: A scope is needed, but I can try to limit the scope to namespaces instead of the root scope.

Comment (venilnoronha) [reply install/wavefront/templates/handler.yaml]: I don't mind leaving it as `.`. I just wanted to confirm.

**Pull request merged**